### ISSUE TITLE
wrong password

### ISSUE THREAD

**Issue by mLgz0rn** [thexmanxyz/network-share-reconnecter#6]
Hey there! so i'm trying to use your script, since i've got the issue with my networkdrive not reconnection.

It does seem to work, but for some reason it tells me the password is wrong, even though i've entered the correct one.

`Set srvCfg1 = createUncSrvConfig("192.168.0.35", Array("Media Server"), Array("Z:"), "yes", "my user", "my password")`

**Comment by thexmanxyz** [comment]
@mLgz0rn Hey, ok that's strange however let us try to isolate the issue you are facing. Am I right that your UNC server share path is `\\192.168.0.35\Media Server`? Does you password contain a blank character?

**Comment by mLgz0rn** [comment]
yup its either \\NVIDIA-SHIELD\Media Server\ or \\192.168.0.35\Media Server\ both work, when I double click the drive it asks me to enter my password.
I do, and then I get fine into the drive.
the password does not contain any blank characters

**Comment by thexmanxyz** [comment]
@mLgz0rn ok that looks ok to me and should therefore be configured right. However please try the following command in command line so that I can check if this works for you. Maybe there is something messed up which I missed.

`net use Z: "\\192.168.0.35\Media Server" /user:YOURUSERNAME YOURPASS /persistent:yes`

You need to provide the full user name which you also use in windows to authenticate with the share. If this connects you with the share correctly it also should with the script if not let me know what error you get.

**Comment by mLgz0rn** [comment]
That works fine, if I use net use Z: /delete and then net use Z: "\\192.168.0.35\Media Server" /user:YOURUSERNAME YOURPASS /persistent:yes the drive maps fine and gets connected

**Comment by mLgz0rn** [comment]
I also just tried using net use Z: /delete and then run your script, and that also works fine.
So guess something goes wrong when I boot and it runs the script

**Comment by thexmanxyz** [comment]
@mLgz0rn And with the manual `net use` the password prompt does not show up? Only with the script?

**Comment by mLgz0rn** [comment]
No, it reconnect to the drive straight away, it dosnt do it either with your script when I try to run it after i've booted into windows

**Comment by thexmanxyz** [comment]
@mLgz0rn Sorry for misconception but I think I'm not getting it right. What was the difference between executing the `net use` and the script execution itself and what exactly does not work. What I understood is that the `net use` command worked correctly but the script prompts a wrong password info. if I understood something wrong please correct me. However I in the meanwhile prepare a debug version so we can verify if there is an issue with the `net use` command generated by the script. But I already verified the net use generation routine and it looks okay.

**Comment by thexmanxyz** [comment]
@mLgz0rn There is another thing that now comes into my mind which I missed. If you have the share always connected and normally never enter a password because you created it already persistent during the share creation. You can skip the password and username section and leave it empty as follows. That should work and that's also how I connect my local network shares. 

`Set srvCfg1 = createUncSrvConfig("192.168.0.35", Array("Media Server"), Array("Z:"), "yes", "", "")`

**Comment by mLgz0rn** [comment]
Okay, starting from the beginning :D 

When I boot my pc from a shutdown, the network drive does get mapped but it is not connect "red cross over it"
When I doubleclick the drive, it prompts me to type in my password, when I do that I get connected to the drive, and can use it as normal.

Now while still in windows without rebooting or shutting down the pc.

I use
`net use Z: /delete`
to delete the drive so I can test.
I then use
`net use Z: "\\192.168.0.35\Media Server" /user:YOURUSERNAME YOURPASS /persistent:yes`
The drive shows up straight away, and does not prompt me to type in my password, and I can use it as normal.

I then do
`net use Z: /delete` 
So I can test if your script is actually working.
I go to where I have your script place, doubleclick it to execute it.
The drive then shows up straight away just like with net use, and does not prompt me to use a password.

It only fails and ask me for a password after a reboot/shutdown when it runs the script through task scheduler

**Comment by mLgz0rn** [comment]
As in, I don't have to type in my username/password when I access the share through "network"?
![network](https://user-images.githubusercontent.com/6141439/36331745-6057a770-136f-11e8-861a-5d4fdd301cb7.PNG)

**Comment by thexmanxyz** [comment]
@mLgz0rn Ok I now fully understand the issue. Thanks for the in detail explanation of the issue. Lets try the following. Connect the drive via explorer save the password or do it with the full net use command I stated above. Modify the script configuration and set it like I wrote it in my last post. Don't specify a `username` nor a `password`. Check if this now works after restart and please report back :). You can also set `debug` to `true` in the script to verify what happens after each connection attempt.

**Comment by mLgz0rn** [comment]
It seems to be working now, thank you so much.
I'll report back if something changes :D

**Comment by thexmanxyz** [comment]
@mLgz0rn Glad to hear that it works now :). I leave the issue open until I verified something because you pointed on a possible issue which I maybe missed and which is probably also the reason why the script didn't work out of the box for you. If I find something out I will reply here otherwise I will close the ticket but if you experience any issue feel free to post again here and I reopen it.

**Comment by thexmanxyz** [comment]
@mLgz0rn I think I found the reason for the issue and why it initially didn't work for you. I also reproduced it on my machine. I will modify the script tomorrow and include a fix that this won't occur anymore. See Issue #7 for more information.

**Comment by mLgz0rn** [comment]
So for some reason unknown, your script stopped working again, havnt worked for a few weeks.
I also tried the newer version you posted, with same results.
there's just a big fat red cross over the drive every time I turn on my pc from a fresh boot

**Comment by thexmanxyz** [comment]
@mLgz0rn First it has to be said that I have no issue on none of my three PCs were the script is in use on a daylie base. 

Going back to the initial issue because I again don't know what situation you are facing in front of your PC. Do you have to enter your credentials after each startup when you click manually on your drive like the last time we wrote? Is the issue exactly the same like the last time?

If yes you can right now download the last version from the repo (not the release). Because there is already a fix applied. The last commit changed the corresponding lines. Didn't create a release version because there are still more thing which need to be done and I didn't had the time to do so.

It might be necessary to enter the credentials within the script. If you still get prompted each time for credentials. But generally I have to say there might be an issue with your Windows installation. That's a thing to consider because credentials shouldn't get lost by the Windows Credential Manager. Only if you change some major things on either your network device, computer or the credentials itself.

**Comment by thexmanxyz** [comment]
@mLgz0rn If everything concerning the issue is the same like the last time we wrote. I recommend the following steps:

1.) download latest repo version
2.) change the script settings
3.) enter your credentials within the script

I have fixed the issue which caused your problems in the first place. So this time there should be no need to waste time on manually doing any `net use`.

**Comment by mLgz0rn** [comment]
It is not the same now. 
It dosnt ask me to input any of my credentials, the drive just has a red cross over it, and I have to doubleclick it to make it active.
I have tried both with and without my credentials but both time its the same. 
As I wrote, I am using the latest repo which included the fix for my first issue

**Comment by mLgz0rn** [comment]
Okay, so I finally got it working.
For some reason I had to use the network name of my shield instead of the ip
`createUncSrvConfig("NVIDIA-SHIELD", Array("Media Server"), Array("Z:"), "yes", "", "")`
it is working now, I also decreased `scriptConfig.reconWait` from 2500 to 1000.

**Comment by thexmanxyz** [comment]
@mLgz0rn Ok I understand the "new" issue now. First short on the problem, it's indeed strange that DN works but the IP does not. Did the IP change in the meantime? I can't explain otherwise than that if the DN works but the IP does not. Windows has more likely problems with DN I never faced any issue with the IP on networks with Windows. Sorry I misunderstood that you already had the latest from the repository. The reconWait can be tweaked to 1000. I used 2500 as a conservative approach but even lower values like 250 or 500 should be no problem at all.

However I will now provide you with some insides on a future release, which also has some relations with your actual issue. It's not always 100% guranteed that a `net use` command indeed establishes full connectivity with a network drive. IDK why this is the case but I know it's the case. It's not very frequent but it can actually happen. In such a case the host/server is available and the `net use` is executed without an error but the drive does not get connected properly. Hence, I'm already thinking about a solution to test the network drive on it's actual connectivity state this would allow me to simply force another `net use` which then would finally establish connectivity. I also verified this in my tests. Such a fallback would finally remove all issues with nw drives. But however I didn't found a way to determine this. But I have to put more investigations on the APIs.

In the current situation this does not really happen for me because the scheduler starts at 10 / 15seconds after login / unlock which prevents the above described problem from happening. I never saw it except if I set the scheduler to 0 / 0 and even then it's 1 out of 10 or so.

**Comment by mLgz0rn** [comment]
@thexmanxyz I was too fast, DN has the same issue as the IP.
I got no clue what's going on.
Could it be related to Fastboot/UEFI boot? 
When I thought I had it working, I shutdown the pc, waited a few seconds and booted it up again, and it worked.
But if I turn of my pc, lets say before I go to bed, and turn it on the next day, the issue will be there again.

I'm sitting on a nvme drive as boot, so it take like 4-5 seconds to boot into windows.

**Comment by thexmanxyz** [comment]
@mLgz0rn I think it has nothing to do with fastboot/uefi. You could in the meantime increase the scheduler time until you hit a spot were the connectivity works for you. I can also create a special fix for you if you want which may fix the problem until I have the check functionality in. Just reply and I will prepare a mod for the script.

**Comment by thexmanxyz** [comment]
@mLgz0rn I created a version for you which forces 3 `net use` on every script execution for each drive. Now it should work. Please check that:

`scriptConfig.netUseCtn = 3`

is set like this. Thanks and I hope we have now a working state.

**Comment by thexmanxyz** [comment]
@mLgz0rn I changed now the script. Please check the attached zip file. I removed some checks and be aware that the script config is set like follows:

`scriptConfig.netUseCtn = 3`

This forces the script to always execute three `net use` commands after some wait time this should circumvent the issue at least as a hacky fix.

PS.: It's not the best way to do it but at least the script should now work and do what it should. I will investigate deeper and improve the existing net use behaviour.

PSS.: I readded the mod forgot to change the `scriptConfig.netUseWait = 500`

[script_mod.zip](https://github.com/thexmanxyz/network-share-reconnecter/files/1852818/script_mod.zip)

**Comment by mLgz0rn** [comment]
@thexmanxyz Thank you so much for your hard work, i'll test it out straight away.

**Comment by mLgz0rn** [comment]
@thexmanxyz Okay, so it seems to be working now! I did a completely cold boot, turned off the psu.

I'm gonna report back tomorrow if it still works or not! :D

**Comment by thexmanxyz** [comment]
@mLgz0rn Glad to hear that it works for the moment. No problem. Thanks also for your patience and motivation of testing :). Please report back if there are any issues and I will so if I finally handle the issue permanently.

**Comment by mLgz0rn** [comment]
@thexmanxyz No luck today when I turned on my pc :/ Still having the issue.

**Comment by thexmanxyz** [comment]
@mLgz0rn Delay the script in the scheduler incrementally until you get a robust state, I don't have a better solution right now. I did all that was for the moment possible. You have to wait on the next release. Because I also can't say why it doesn't work for you.

You can also set `scriptConfig.netUseCtn = 10` or higher maybe you find a sweet spot. However this are just cheap workarounds...

**Comment by thexmanxyz** [comment]
@mLgz0rn Another thing I have to mention is, that it might be that the Explorer displays the network drives with a red cross but they are actually connected. Wait a bit after restart and close / reopen Explorer. The icon indication does not update in real time if explorer is open. Just as an idea which came right now in my mind. I didn't mention that.

**Comment by mLgz0rn** [comment]
Such a long time! but I might finally have figured out what the issue is.
The problem is that the Nvidia Shield TV that im running the smb server on is only running smb v1.
A few threads I have found, says that smb v1 has a few issues with remounting/reconnecting a networkdrive on windows 10. Which is also some of the reasons why microsoft has discontinued smb v1 and hidden it in w10.
The problem is that at the moment the shield tv does not support smb v2 or v3. But the devs has promised that it will be added in one of the next updates.
